### PR TITLE
DAOS-13013 test: Adding replay test case with no checkpointing

### PR DESCRIPTION
**Pull request by phender**
Adding a test case to verify data access after engine restart with WAL replay and no check pointing.

Skip-unit-tests: true
Test-tag: ReplayTests
Test-nvme: auto_md_on_ssd

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/server/replay.py [hunk 1]
```diff
@@ -326,3 +326,46 @@ def test_replay_attributes(self):
...
+        container.pool.set_prop(properties='checkpoint:disabled')
+        response = container.pool.get_prop(name='checkpoint')['response']
+        if response[0]['value'] != 'disabled':
+            self.fail('Pool check pointing not disabled before engine restart')
+
+        self.log_step('Write data to the container (ior)')
+        ior = self.write_data(container, ppn)
+
+        self.stop_engines()
+        self.restart_engines()
```
Comment (daltonbohning) [inline src/tests/ftest/server/replay.py:360]: Did you intend to put a `log_step` here?

**Review by daltonbohning — APPROVED** [review]
LGTM

File: src/tests/ftest/server/replay.py (hunk 1 above)
Comment (phender) [reply src/tests/ftest/server/replay.py:360]: Its handled by the methods:

- https://github.com/daos-stack/daos/blob/79cf11dc86d235302492d91d53e25c58669d6c59/src/tests/ftest/server/replay.py#L57
- https://github.com/daos-stack/daos/blob/79cf11dc86d235302492d91d53e25c58669d6c59/src/tests/ftest/server/replay.py#L69

**Comment by phender** [comment]
All tests passed in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-11959/1/testReport/FTEST_server/ReplayTests/

**Review by dinghwah — APPROVED** [review]

**Pull request merged**